source/docs/software/wpilib-tools/wpical/index.rst

```diff
@@ -0,0 +1,192 @@
+
+Overview
+--------
+Conceptually, if you know how far tags are away from each other, and you know the field-relative coordinates of one of the tags, (the "pinned" tag) you can figure out the field-relative coordinates of the other tags. WPIcal achieves this by running an optimization on frames from videos that the user takes of the field. WPIcal is field-agnostic, meaning it will work with any FRC apriltag layout, as well as custom apriltag layouts, as long as they follow a couple of rules:
+
+* Translation components (in meters) are measured relative to the blue alliance origin.
+* Rotations are represented as Quaternions.
+
+Example:
+
+.. code-block:: json
+
+    {
+        "ID": 1,
+        "pose": {
+            "translation": {
+                "x": 15.079471999999997,
+                "y": 0.24587199999999998,
+                "z": 1.355852
+            },
+            "rotation": {
+                "quaternion": {
+                    "W": 0.5000000000000001,
+                    "X": 0.0,
+                    "Y": 0.0,
+                    "Z": 0.8660254037844386
+                }
+            },
+        },
+    }
+
+.. important:: WPIcal is meant to correct for SMALL variations in tag placement. It is still important that you set up your Apriltags in mostly the correct location and orientation, so WPIcal performs the optimal calibration.
+.. important:: Make sure that you verify the results of each calibration thoroughly to ensure that your calibration matches your field setup accurately.
+
+Camera Calibration
+------------------
+To measure the distance between tags in a video, you need the camera's intrinsics to account for things like distortion. WPIcal allows users to upload a video file to calibrate the camera in the tool, or upload a JSON file with your camera's intrinsics
+
+In-App Calibration
+^^^^^^^^^^^^^^^^^^
+To calibrate your camera from a video file, click on :guilabel:`Calibrate Camera`
+
+.. image:: images/Calibrate.png
+    :alt: Calibrate
+
+There are two options for calibrating your camera, :guilabel:`OpenCV` and :guilabel:`MRcal`. It is generally recommended to calibrate with :guilabel:`MRcal`, as it has slightly better accuracy and is more robust against bad calibration data. WPIcal uses a ChArUco board for camera calibration. You can generate a ChArUco board here: `calib.io <https://calib.io/pages/camera-calibration-pattern-generator>`_
```

Comment:
Any particular recommendations on board width/height? I noticed the gui defaults to 12x8. And whitch dictionary should I be using?

Reply:
(wpical seems to be hard-coded to 5x5?)

Reply:
Yes the calibration is hard coded to 5x5. I was testing with a 12x8 board and it seemed to work well. I haven't tested other widths or heights too thoroughly. The optimal dimensions will vary camera to camera right?